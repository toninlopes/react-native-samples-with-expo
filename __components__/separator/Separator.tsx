import React from "react";
import { ColorValue, StyleSheet, View } from "react-native";
import type { SeparatorProps } from './types';

const Separator: React.FC<SeparatorProps> = ({ color, style }) => {
  return (
    <View style={[styles().separator, styles(color, style).border]} />
  );
};

const styles = (
  color?: ColorValue | undefined,
  style?: 'solid' | 'dotted' | 'dashed' | undefined,
) => ({
  ...StyleSheet.create({
    separator: {
      flex: 1,
    },
    border: {
      borderWidth: 0.8,
      borderColor: color || 'silver',
      borderStyle: style || 'dashed',
    },
  }),
});

export default Separator;