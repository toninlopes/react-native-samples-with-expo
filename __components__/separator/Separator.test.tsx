import React from 'react';
import renderer from 'react-test-renderer';

import Separator from './Separator';

describe('Separator', () => {
  test('When there is color.', () => {
    const button = renderer.create((
      <Separator color={'black'} />
    )).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('When there is style.', () => {
    const buttonWithSolidStyle = renderer.create((
      <Separator style='solid' />
    )).toJSON();
    expect(buttonWithSolidStyle).toMatchSnapshot();

    const buttonWithDashedStyle = renderer.create((
      <Separator style='dashed' />
    )).toJSON();
    expect(buttonWithDashedStyle).toMatchSnapshot();

    const buttonWithDottedStyle = renderer.create((
      <Separator style='dotted' />
    )).toJSON();
    expect(buttonWithDottedStyle).toMatchSnapshot();
  });
});