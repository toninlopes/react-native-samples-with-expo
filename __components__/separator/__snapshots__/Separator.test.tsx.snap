// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator When there is color. 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderColor": "black",
        "borderStyle": "dashed",
        "borderWidth": 0.8,
      },
    ]
  }
/>
`;

exports[`Separator When there is style. 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderColor": "silver",
        "borderStyle": "solid",
        "borderWidth": 0.8,
      },
    ]
  }
/>
`;

exports[`Separator When there is style. 2`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderColor": "silver",
        "borderStyle": "dashed",
        "borderWidth": 0.8,
      },
    ]
  }
/>
`;

exports[`Separator When there is style. 3`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderColor": "silver",
        "borderStyle": "dotted",
        "borderWidth": 0.8,
      },
    ]
  }
/>
`;
